WatchFaceAlphaKotlin/app/src/main/java/com/example/android/wearable/alpha/editor/WatchFaceConfigViewModel.kt

```diff
@@ -0,0 +1,265 @@
+ */
+class WatchFaceConfigViewModel(activity: ComponentActivity, editIntent: Intent) : ViewModel() {
+    private val changeSupport = PropertyChangeSupport(this)
+
+    private fun addPropertyChangeListener(listener: PropertyChangeListener) {
+        changeSupport.addPropertyChangeListener(listener)
+    }
+
+    private fun removePropertyChangeListener(listener: PropertyChangeListener) {
+        changeSupport.removePropertyChangeListener(listener)
+    }
+
+    private lateinit var editorSession: EditorSession
+
+    // Watch Face styles the user can actively edit
+    private lateinit var userStyle: UserStyle
+
+    // Preview of complication data (needed to render screenshots)
+    private lateinit var previewComplicationData: Map<Int, ComplicationData>
+
+    // Keys from Watch Face Data Structure
+    private lateinit var colorStyleKey: UserStyleSetting.ListUserStyleSetting
+    private lateinit var drawPipsKey: UserStyleSetting.BooleanUserStyleSetting
+    private lateinit var minuteHandLengthKey: UserStyleSetting.DoubleRangeUserStyleSetting
+
+    // The UI collects from this StateFlow to get its state updates
+    @OptIn(ExperimentalCoroutinesApi::class)
+    val uiState: StateFlow<EditWatchFaceUiState> = callbackFlow {
+        val propertyChangeListener = PropertyChangeListener { event ->
+            Log.d(TAG, "Change listener: ${event.propertyName}")
+            trySend(event.newValue as UserStylesAndPreview)
+        }
+        addPropertyChangeListener(listener = propertyChangeListener)
+
+        // The callback inside awaitClose will be executed when the flow is either closed or
+        // cancelled. In this case, remove the CurrentUserStyleRepository.UserStyleChangeListener()
+        // callback.
+        awaitClose {
+            Log.d(TAG, "awaitClose{ }")
+            removePropertyChangeListener(propertyChangeListener)
+        }
+    }
+        .buffer(Channel.CONFLATED)
+        .map(EditWatchFaceUiState::Success)
+        .stateIn(
+            scope = viewModelScope,
+            started = SharingStarted.Eagerly,
+            initialValue = EditWatchFaceUiState.Loading("Initializing")
+        )
+
+    init {
+        viewModelScope.launch(Dispatchers.Main.immediate) {
+
+            editorSession = EditorSession.createOnWatchEditorSession(
+                activity = activity,
+                editIntent = editIntent
+            )
+
+            previewComplicationData = editorSession.getComplicationsPreviewData()
+            userStyle = editorSession.userStyle
+            Log.d(WatchFaceConfigActivity.TAG, "userStyle: $userStyle")
+
+            extractsUserStyles(userStyle)
+            updatesWatchFacePreview()
+        }
+    }
+
+    private fun extractsUserStyles(newUserStyle: UserStyle) {
+        // Loops through user styles and retrieves user editable styles.
+        for (options: Map.Entry<UserStyleSetting, UserStyleSetting.Option> in newUserStyle.selectedOptions) {
```

Comment:
Nit: I think you don't need to specify the type here unless you want to do so for clarity.